localstack/services/sqs/provider.py

```diff
@@ -1069,16 +1069,20 @@ def _validate_actions(self, actions: ActionNameList):
                     "WSDL for a list of valid actions. "
                 )
 
-    def _assert_batch(self, batch: List):
+    def _assert_batch(self, batch: List) -> None:
         if not batch:
             raise EmptyBatchRequest
         visited = set()
         for entry in batch:
-            # TODO: InvalidBatchEntryId
-            if entry["Id"] in visited:
+            if not re.search(r"^[\w-]+$", (entry_id := entry["Id"])) or len(entry_id) > 80:
```

Comment:
nit: took me a while to see where `entry_id` is coming from. i think it's better not to use the implicit assignment. i think the extra line makes the code more readable.
```suggestion
            entry_id = entry["Id"]
            if not re.search(r"^[\w-]+$", entry_id) or len(entry_id) > 80:
```